typescript/lib/parsers/palm.ts

```diff
@@ -29,22 +69,75 @@ export class PaLMTextParser extends ParameterizedModelParser {
     options?: InferenceOptions | undefined,
     params?: JSONObject | undefined
   ): Promise<Output[]> {
-    throw new Error("Method not implemented.");
+    if (!this.client) {
+      const apiKey = getAPIKeyFromEnv("PALM_KEY");
+      this.client = new TextServiceClient({
+        authClient: new GoogleAuth().fromAPIKey(apiKey),
+      });
+    }
+
+    const completionParams = this.deserialize(prompt, aiConfig, params);
+    // Palm doesn't support streaming
```

Comment:
nit: add API url

Reply:
added api docs at the top. added comment to clarify